File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int b = 0, e = nums.length - 1, m = 0;
        while(b < e) {
            m = (b + e) / 2;
            if(m % 2 == 0 && nums[m] == nums[m+1] || m % 2 == 1 && nums[m] == nums[m-1]) {
                b = m + 1;
            }
            else {
                e = m;
            }
        }
        return nums[b];
    }
}

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int i = 0;
        while(i < nums.size()) {
            if(i != nums.size() - 1 and nums[i] == nums[i+1]) {
                i += 2;
            }
            else {
                return nums[i];
            }
        }
        return nums[i];
    }
};